### DISCUSSION TITLE
v4l2-ctl, unreasonable fps output

### DISCUSSION THREAD

**Discussion by danny1209** [OE4T/tegra-demo-distro#234] [Q&A]
I used this command to capture images:
v4l2-ctl -d /dev/video0 --set-fmt-video=width=1792,height=1204 --set-ctrl bypass_mode=0,override_enable=1,sensor_mode=1,gain=2048,exposure=4000,frame_rate=25000000,illumination=1,low_latency_mode=1 --stream-mmap --stream-count=100 --stream-to=nov3_25fps_raw12_ram.raw

in Jetpack the output is ok:
<<<<<<<<<<<<<<<<<<<<<<<<<<< 26.00 fps
<<<<<<<<<<<<<<<<<<<<<<<<<<< 26.23 fps
<<<<<<<<<<<<<<<<<<<<<<<<<<< 26.40 fps
<<<<<<<<<<<<<<<<<<<

but in yocto seems unreasonable:
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< 1025.61 fps, dropped buffers: 1009
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< 1003.10 fps, dropped buffers: 945
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< 990.49 fps, dropped buffers: 932
<<<<<<<

does somebody have the same problem?

**Comment by ichergui** [comment]
Hello @danny1209 
Are you using the tegra demo distro ? If so, please share the branch name 
Also, are you using the same video source input ? Check the video resolution and the frame rate

**Comment by danny1209** [comment]
Hi @ichergui 
I´m using this one: https://github.com/OE4T/tegra-demo-distro/tree/dunfell
yes the same video source, same hardware and alle same settings.

**Reply by ichergui** [reply]
Which machine you are using ?
I didn't get issue similar to yours 
I will check again

**Reply by danny1209** [reply]
jetson nano

**Comment by madisongh** [comment]
Do you get the same result if you disable low-latency mode, and which version of JetPack are you running?  There appears to be a bug in the vi driver's handling of low-latency mode, as mentioned [here](https://forums.developer.nvidia.com/t/221208/7/). It sounds like the bug was introduced in JetPack 4.6.1 (L4T R32.7.x), which is what our dunfell branch is based on.  If the JetPack you're running is older, you would not see this bug.

**Reply by danny1209** [reply]
Hi @madisongh 
thanks for the reply.
yes, it does resolve the output issue by setting low_latency_mode=0. Now the output looks reasonable. but we are prefer to enable low_latency_mode. 
I´m using JetPack 4.6.2. I thought dunfell branch is also based on 4.6.2, Is it correct? 
![image](https://user-images.githubusercontent.com/46848895/201623442-53e60f68-d4cc-4b96-a18a-a2bb6e7d54a9.png)

**Reply by danny1209** [reply]
I do also have the image sequence issue. 
my camera has 4 contexts, which I can make different settings for each context and get individual image, but it happens often, the first image from context0 is just missing.

**Reply by madisongh** [reply]
4.6.1 (L4T R32.7.1) and 4.6.2 (L4T R32.7.2) should behave the same, since there were no kernel changes as part of that update.

**Reply by danny1209** [reply]
ok, I´m using JetPack4.6.2 on another device, don´t see the bug about unreasonable fps output.

**Reply by madisongh** [reply]
Hmm, OK.  Perhaps you could enable some of the kernel debug settings that were mentioned in that forum post in both environments and compare the output to try and figure out what's going on?
